src/marqo/marqo_cloud_instance_mappings.py

```diff
@@ -1,37 +1,48 @@
 import time
-from marqo.marqo_logging import mq_logger
+from typing import Optional
+
 import requests
 from requests.exceptions import Timeout
+
 from marqo.errors import (
     MarqoCloudIndexNotFoundError,
     MarqoCloudIndexNotReadyError,
 )
+from marqo.instance_mappings import InstanceMappings
+from marqo.marqo_logging import mq_logger
+
 
+class MarqoCloudInstanceMappings(InstanceMappings):
+    _MARQO_CLOUD_URL = "https://api.marqo.ai"
 
-class MarqoUrlResolver:
     def __init__(self, api_key=None, expiration_time: int = 15):
-        """ URL Resolver is a cache for urls that are resolved to their respective indices only for marqo cloud. """
         self.timestamp = time.time() - expiration_time
```

Comment:
Can we also rename `self.timestamp` to something more descriptive? For example, something like `self.latest_index_mappings_refresh_timestamp`

Reply:
agreed